src/language/classes.md

```diff
@@ -175,10 +175,10 @@ Non-final instance variables and
 an implicit *setter* method. For details,
 check out [Getters and setters][].
 
-If you initialize a non-`late` instance variable where it's declared,
+If you initialize a non-`late` `final` instance variable where it's declared,
```

Comment:
I'm not sure why you'd think that this doesn't initialize `i`:

```dart
class A {
  int i = 1;
}

void main() {
  print(A().i); // '1'.
}
```

Reply:
Respected,

I am talking about non-late but final instance variable which is initialized at declaration time. In that case "this" keyword will not be accessible.

```
class Point{
final double a;
final double b;

 //initialized at declaration
final double c=2.7;

Point(this.a,this.b,this.c);

}

```
In this case "this.c" will show error. Because it is already initialized.

But if "c" is not final then you can reassign it a new value.

Reply:
If I understand correctly, @pdhankhar61 you're issue is with the existing statement:

> non-late instance variable initializers can’t access this.

And you're saying that statement is incorrect because you tried accessing `this` for the non-late instance variables `a`, `b` and `c`, and it works:

```dart
class Point{
double a;
double b;

 //initialized at declaration
double c=2.7;

Point(this.a,this.b,this.c); // no errors, accessing `this` DOES work even though c is already initialized
}
```

But you found that statement *is* true if the variable is `final`, so you're updating the statement to specify `final`?

```dart
class Point{
final double a;
final double b;

 //initialized at declaration
final double c=2.7;

Point(this.a,this.b,this.c); // error - 'c' is final and was given a value when it was declared, so it can't be set to a new value.
}
```

I could be totally missing the point of the section, but I'm finding it confusing too. It also doesn't seem like the example code starting on line 183 is related to the paragraph immediately preceding it (the same paragraph in question), which makes it even more confusing. 

@eernstg (or anyone) So it's not just about initialization of non-late instance variables in general, but specifically that the paragraph makes it sound like you can't use `this` to initialize non-late instance variables -- why does it say that doesn't work?

Reply:
Respected,

Yes, you understood it correctly.

Reply:
Okay Now, I have understood you point @eernstg sir. Like that statement was confusing without example. 

Delay in response is because of github, because it sends notifications with a delay.

Reply:
Thanks, @pdhankhar61!

So what have we learned here? Taking a look at [the section](https://github.com/dart-lang/site-www/blob/4ff61a8486a80b14b8d5153b1b88c8693a942df7/src/language/classes.md#instance-variables) as a whole, the following comes to mind:

I think it's confusing that this section says 'All uninitialized instance variables have the value `null`', and it does not even mention that for many instance variables (namely the ones whose declared type is potentially non-nullable) it is simply a compile-time error if they are left uninitialized.

If we mention this then it also makes sense to mention the error, perhaps right after "can't access `this`":

```dart
double initialX = 1.5;

class Point {
  double? x = initialX; // OK, can access declarations that do not depend on `this`.
  double? y = this.x; // Compile-time error, cannot access `this`.
}
```

@MaryaBelanger, I understand that this kind of documentation should be extremely succinct and to the point, but what do you think about adding something like this?

We should probably also mention somewhere that an initializing formal isn't an expression and hence it isn't an "access to `this`" even in the case where it's written literally as `this.x`, but that might be better placed in a section where initializing formals are introduced (the same section does use one initializing formal, but only refers to it as 'a constructor parameter', so perhaps initializing formals aren't described here at all).